### PR TITLE
Add GeneratedCodeAttribute tests (System.Diagnostics.Tools partial facade)

### PR DESCRIPTION
**Pull request by mellinoe**
Fills in code coverage for this partial facade. We can add additional tests if/when we move additional types out into the partial facade.

### CODE REVIEW

**Comment by mellinoe** [comment]
Seems that we got this in the CI build:

``d:\j\workspace\dotnet_corefx_linux_release_prtest\packages\Microsoft.DotNet.BuildTools.1.0.25-prerelease-00055\lib\packageresolve.targets(26,5): error MSB3073: The command ""d:\j\workspace\dotnet_corefx_linux_release_prtest\packages//dnx-coreclr-win-x86.1.0.0-beta5-12101/\bin\dnu.cmd" restore --packages "d:\j\workspace\dotnet_corefx_linux_release_prtest\packages" "d:\j\workspace\dotnet_corefx_linux_release_prtest\src\System.Diagnostics.Debug\src/project.json"" exited with code 1. [d:\j\workspace\dotnet_corefx_linux_release_prtest\src\System.Diagnostics.Debug\src\System.Diagnostics.Debug.csproj]``

File: src/System.Diagnostics.Tools/tests/System.Diagnostics.Tools.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <OutputItemType>Content</OutputItemType>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      <Targets>Build;DebugSymbolsProjectOutputGroup</Targets>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="project.json" />
+  </ItemGroup>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+</Project>
```
Comment (stephentoub) [inline src/System.Diagnostics.Tools/tests/System.Diagnostics.Tools.Tests.csproj]: nit: missing newline

File: src/System.Diagnostics.Tools/tests/System/CodeDom/Compiler/GeneratedCodeAttributeTests.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+namespace System.CodeDom.Compiler
+{
+    public static class GeneratedCodeAttributeTests
+    {
+        [Fact]
+        public static void Constructor()
+        {
+            string tool = "Tool";
+            string version = "Version";
+            GeneratedCodeAttribute gca = new GeneratedCodeAttribute(tool, version);
```
Comment (stephentoub) [inline src/System.Diagnostics.Tools/tests/System/CodeDom/Compiler/GeneratedCodeAttributeTests.cs]: Maybe make this a theory and check that nulls pass through fine, e.g.
```C#
[Theory]
[InlineData(null, null)]
[InlineData("Tool", "Version")]
public void Constructor(string tool, string version)
{
    var gca = new GeneratedCodeAttribute(tool, version);
    Assert.Equal(tool, gc.Tool);
    Assert.Equal(version, gc.Version);
}
```

**Comment by stephentoub** [comment]
Two minor comments, otherwise LGTM. (Again the with same comment as on System.IO about assuming we'll be adding tests soon for the mscorlib-portion of the assembly.)

File: src/System.Diagnostics.Tools/tests/System.Diagnostics.Tools.Tests.csproj (hunk 1 above)
Comment (mellinoe) [inline src/System.Diagnostics.Tools/tests/System.Diagnostics.Tools.Tests.csproj]: VS automatically deletes the newline at the end of the file

**Comment by mellinoe** [comment]
Looks like the Linux PR test build just took too long and ended up getting killed? Have we seen this before?

``Build timed out (after 120 minutes). Marking the build as aborted.``

**Comment by stephentoub** [comment]
I believe we've seen it when dnx hangs.... @mmitche?

**Comment by mmitche** [comment]
Yep.  Looks like the last dnx update seems to have upped the frequency of this.

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Pull request merged**